common/src/sql/dbinit.sql

```diff
@@ -1419,6 +1419,98 @@ CREATE INDEX ON omicron.public.update_available_artifact (
     targets_role_version
 );
 
+/*
+ * System updates
+ */
+CREATE TABLE omicron.public.system_update (
+    /* Unique semver version */
+    version STRING(64) PRIMARY KEY,
+
+    -- The ID is not strictly necessary here because we're using the version
+    -- string as the PK. However, removing it would make this one of the only
+    -- resources without an ID, which means we'd have to do a bunch of special
+    -- stuff that we get more or less for free with other resources. For
+    -- example, paginating by ID (note this is distinct from sorting by ID).
+
+    /* Identity metadata (asset) */
+    id UUID NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL
+);
+
+ 
+-- Used for the join with components. That and pagination is all id is used for
+CREATE UNIQUE INDEX ON omicron.public.system_update (
+    id
+);
+
+CREATE TYPE omicron.public.updateable_component_type AS ENUM (
+    'bootloader_for_rot',
+    'bootloader_for_sp',
+    'bootloader_for_host_proc',
+    'hubris_for_psc_rot',
+    'hubris_for_psc_sp',
+    'hubris_for_sidecar_rot',
+    'hubris_for_sidecar_sp',
+    'hubris_for_gimlet_rot',
+    'hubris_for_gimlet_sp',
+    'helios_host_phase_1',
+    'helios_host_phase_2',
+    'host_omicron'
```

Comment:
We're probably gonna add zones-updateable-separately-from-the-host-OS like CRDB + Clickhouse to this list, right? (Not in this PR, just kinda thinking out loud)

Reply:
Hadn't thought about it, but yeah, sounds right. I just copied these from the example tree in [RFD 334](https://rfd.shared.oxide.computer/rfd/0334#_requirements).